test/unit/ui/control/geolocate.test.js

```diff
@@ -0,0 +1,299 @@
+    });
+    geolocate._geolocateButton.dispatchEvent(click);
+    geolocation.send({latitude: 10, longitude: 20, accuracy: 1});
+});
+
+test('GeolocateControl no watching map camera on geolocation', (t) => {
+    t.plan(6);
+
+    const map = createMap();
+    const geolocate = new GeolocateControl({
+        fitBoundsOptions: {
+            maxZoom: 20,
+            linear: true,
+            duration: 0
+        }
+    });
+    map.addControl(geolocate);
+
+    const click = new window.Event('click');
+
+    map.on('moveend', () => {
+        t.deepEqual(lngLatAsFixed(map.getCenter(), 4), { lat: 10, lng: 20 }, 'map centered on location');
+
+        const mapBounds = map.getBounds();
+
+        // map bounds should contain or equal accuracy bounds, that is the ensure accuracy bounds doesn't fall outside the map bounds
+        const accuracyBounds = map.getCenter().toBounds(1000);
+        t.ok(accuracyBounds.getNorth().toFixed(4) <= mapBounds.getNorth().toFixed(4), 'map contains north of accuracy radius');
+        t.ok(accuracyBounds.getSouth().toFixed(4) >= mapBounds.getSouth().toFixed(4), 'map contains south of accuracy radius');
+        t.ok(accuracyBounds.getEast().toFixed(4) <= mapBounds.getEast().toFixed(4), 'map contains east of accuracy radius');
+        t.ok(accuracyBounds.getWest().toFixed(4) >= mapBounds.getWest().toFixed(4), 'map contains west of accuracy radius');
+
+        // map bounds should not be too much bigger on all edges of the accuracy bounds (this test will only work for an orthogonal bearing),
+        // ensures map bounds does not contain buffered accuracy bounds, as if it does there is too much gap around the accuracy bounds
+        const bufferedAccuracyBounds = map.getCenter().toBounds(1100);
+        t.notOk(
+            (bufferedAccuracyBounds.getNorth().toFixed(4) < mapBounds.getNorth().toFixed(4)) &&
+            (bufferedAccuracyBounds.getSouth().toFixed(4) > mapBounds.getSouth().toFixed(4)) &&
+            (bufferedAccuracyBounds.getEast().toFixed(4) < mapBounds.getEast().toFixed(4)) &&
+            (bufferedAccuracyBounds.getWest().toFixed(4) > mapBounds.getWest().toFixed(4)),
+        'map bounds is much is larger than the accuracy radius');
+
+        t.end();
+    });
+    geolocate._geolocateButton.dispatchEvent(click);
+    geolocation.send({latitude: 10, longitude: 20, accuracy: 1000});
+});
+
+test('GeolocateControl watching map updates recenter on location with dot', (t) => {
+    t.plan(6);
+
+    const map = createMap();
+    const geolocate = new GeolocateControl({
+        trackUserLocation: true,
+        showUserLocation: true,
+        fitBoundsOptions: {
+            linear: true,
+            duration: 0
+        }
+    });
+    map.addControl(geolocate);
+
+    const click = new window.Event('click');
+
+    let moveendCount = 0;
+    map.once('moveend', () => {
+        // moveend was being called a second time, this ensures that we don't run the tests a second time
```

Comment:
yes that is what I observed. I couldn't work out why that was happening so I just worked around it with this count.